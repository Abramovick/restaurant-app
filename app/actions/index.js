import * as types from '../constants/Actiontypes';

const receiveProducts = products => ({
  type: types.RECEIVE_PRODUCTS,
  products
});

const addToCartFn = product => ({
  type: types.ADD_TO_CART,
  product
});

const addToken = token => ({
  type: types.ADD_TOKEN,
  token
});

const login = loginState => ({
  type: types.LOGIN_FAIL,
  loginState
});

const checkOut = () => ({
  type: types.CHECKOUT,
});

const increaseQuantity = product => ({
  type: types.INCREASE_ITEM_QUANTITY,
  product
});

const decreaseQuantity = product => ({
  type: types.DECREASE_ITEM_QUANTITY,
  product
});

const clearAuth = () => ({
  type: types.CLEAR_AUTH,
});

export const addProducts = products => (dispatch) => {
  dispatch(receiveProducts(products));
};
export const addTokenToStore = token => (dispatch) => {
  dispatch(addToken(token));
};
export const addToCart = product => (dispatch) => {
  dispatch(addToCartFn(product));
};

export const increaseItemQuantity = product => (dispatch) => {
  product.quantity += 1;
  dispatch(increaseQuantity(product));
};

export const decreaseItemQuantity = product => (dispatch) => {
  product.quantity -= 1;
  dispatch(decreaseQuantity(product));
};

export const clearCart = () => (dispatch) => {
  dispatch(checkOut());
};

export const removeAuth = () => (dispatch) => {
  dispatch(clearAuth());
};

export const loginFail = loginState => (dispatch) => {
  dispatch(login(loginState));
};
