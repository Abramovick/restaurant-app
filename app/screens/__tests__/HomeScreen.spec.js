import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { shallow, } from 'enzyme';
import thunk from 'redux-thunk';
import ConnectedHome, { HomeScreen } from '../HomeScreen';
import { addProducts, addTokenToStore, loginFail } from '../../actions';

// Mocked component
jest.mock('redux-form/lib/Field', () => 'Field');

// Test objects
const navigation = {
  navigate: jest.fn()
};

const items = [{
  name: 'mariam',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world '
},
{
  name: 'tope',
  ImageUrl: 'gdfgds',
  Price: 700,
  description: 'the hello is in the world'
}];

const products = [{
  name: 'rice and beans',
  description: 'the rice and beans is dsldks'
}];

const state = {
  cart: {
    items
  },
  user: {
    loginState: true
  }
};

const productQuery = {
  allProducts: {}
};

const values = {
  email: 'mariam3105@gmail.com',
  password: 'rw840424'
};

const response = {
  data: {
    authenticateUser: {
      token: 12345678,
      id: 534534543543
    }
  }
};

// Mocked functions
const rejectedUserMutation = () => Promise.reject(new Error('not found'));
const resolvedUserMutation = () => response;
const handleSubmit = jest.fn();
const LoginFail = jest.fn();

// Test suites
describe('HOMESCREEN --- Snapshot', () => {
  it('captures snapshot of homescreen with rejected auth promise', async () => {
    const tree = renderer.create(<HomeScreen
    state={state}
    loginState
    loginFail={LoginFail}
    handleSubmit={handleSubmit}
    navigation={navigation}
    addProducts={addProducts}
    authenticateUserMutation={rejectedUserMutation}
    addTokenToStore={addTokenToStore}
    productQuery={productQuery}
    />);
    expect(tree.getInstance().loginUser(values)).toMatchSnapshot();
    expect(tree.getInstance().storeAuthTokensLocally()).toMatchSnapshot();
    expect(tree.getInstance().handleSignupPress()).toMatchSnapshot();
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('captures snapshot of homescreen with resolved auth promise', async () => {
    const tree = renderer.create(<HomeScreen
    state={state}
    loginState
    loginFail={LoginFail}
    handleSubmit={handleSubmit}
    navigation={navigation}
    authenticateUserMutation={resolvedUserMutation}
/>);
    expect(tree.getInstance().loginUser(values)).toMatchSnapshot();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('HOMESCREEN --- Shallow rendering + passing the store directly', () => {
  const initialState = {
    cart: {
      items
    },
    user: {
      loginState: true
    }
  };
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  let store, container;

  beforeEach(() => {
    store = mockStore(initialState);
    container = shallow(<ConnectedHome store={store}/>);
  });

  it('renders the connected component', () => {
    expect(container.length).toEqual(1);
  });

  it('check Prop matches with initialState', () => {
    expect(container.prop('loginState'))
      .toEqual(initialState.user.loginState);
  });
  it('check action on dispatching ', () => {
    let action = '';
    store.dispatch(addProducts(products));
    store.dispatch(addTokenToStore('klmkmkl67890'));
    store.dispatch(loginFail(true));
    action = store.getActions();
    expect(action[0].type).toBe('RECEIVE_PRODUCTS');
    expect(action[1].type).toBe('ADD_TOKEN');
    expect(action[2].type).toBe('LOGIN_FAIL');
  });
});
