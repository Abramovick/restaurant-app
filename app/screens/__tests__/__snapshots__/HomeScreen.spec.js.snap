// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with rejected auth promise 1`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with rejected auth promise 2`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with rejected auth promise 3`] = `undefined`;

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with rejected auth promise 4`] = `
<View
  style={
    Object {
      "backgroundColor": "#D57E56",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#D57E56",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 30,
              "marginTop": "9%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontFamily": "SinkinSans-100Thin",
                  "fontSize": 30,
                },
              ]
            }
          >
            Sign  In
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "#F7C04C",
                  "fontFamily": "SinkinSans-100Thin",
                  "fontSize": 18,
                },
              ]
            }
          >
            Restaurant
          </Text>
        </View>
        <Field
          component={[Function]}
          name="email"
          placeholder="E-mail"
        />
        <Field
          component={[Function]}
          name="password"
          placeholder="Password"
          secureTextEntry={true}
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginTop": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "red",
                  "fontSize": 12,
                  "marginLeft": 10,
                },
              ]
            }
          >
            Invalid Username or Password
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
                Login
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "SinkinSans-200XLight",
                "fontSize": 12,
              },
              Object {
                "color": "white",
                "fontSize": 12,
                "marginTop": 20,
              },
            ]
          }
        >
          Forgot Password?
        </Text>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 12,
                    "marginTop": 20,
                  },
                ]
              }
            >
              Do not have account?  
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#F7C04C",
                      "fontSize": 12,
                      "marginTop": 20,
                    },
                  ]
                }
              >
                Sign Up here
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#BEB5B2",
                "borderBottomWidth": 0.5,
                "margin": 7,
                "width": 100,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                  "marginTop": 20,
                },
              ]
            }
          >
            Or Log In with
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#BEB5B2",
                "borderBottomWidth": 0.5,
                "margin": 7,
                "width": 100,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "marginHorizontal": 30,
            "marginTop": "8%",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#445EA9",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 0,
              "paddingTop": 0,
              "width": "50%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
              Facebook
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#C6563B",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "width": "50%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
              Google
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with resolved auth promise 1`] = `
Promise {
  "_40": 0,
  "_55": null,
  "_65": 0,
  "_72": null,
}
`;

exports[`HOMESCREEN --- Snapshot captures snapshot of homescreen with resolved auth promise 2`] = `
<View
  style={
    Object {
      "backgroundColor": "#D57E56",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingTop": 30,
      }
    }
    style={
      Object {
        "backgroundColor": "#D57E56",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginHorizontal": 30,
          }
        }
      >
        <View
          style={
            Object {
              "marginBottom": 30,
              "marginTop": "9%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontFamily": "SinkinSans-100Thin",
                  "fontSize": 30,
                },
              ]
            }
          >
            Sign  In
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "#F7C04C",
                  "fontFamily": "SinkinSans-100Thin",
                  "fontSize": 18,
                },
              ]
            }
          >
            Restaurant
          </Text>
        </View>
        <Field
          component={[Function]}
          name="email"
          placeholder="E-mail"
        />
        <Field
          component={[Function]}
          name="password"
          placeholder="Password"
          secureTextEntry={true}
        />
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
              "marginTop": 5,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "red",
                    "fontSize": 14,
                    "marginLeft": 10,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "red",
                  "fontSize": 12,
                  "marginLeft": 10,
                },
              ]
            }
          >
            Invalid Username or Password
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#F7C04C",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 40,
              "justifyContent": "center",
              "marginTop": "4%",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
                Login
          </Text>
        </View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "SinkinSans-200XLight",
                "fontSize": 12,
              },
              Object {
                "color": "white",
                "fontSize": 12,
                "marginTop": 20,
              },
            ]
          }
        >
          Forgot Password?
        </Text>
        <View>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "SinkinSans-200XLight",
                    "fontSize": 12,
                  },
                  Object {
                    "color": "white",
                    "fontSize": 12,
                    "marginTop": 20,
                  },
                ]
              }
            >
              Do not have account?  
            </Text>
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "fontFamily": "SinkinSans-200XLight",
                      "fontSize": 12,
                    },
                    Object {
                      "color": "#F7C04C",
                      "fontSize": 12,
                      "marginTop": 20,
                    },
                  ]
                }
              >
                Sign Up here
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "borderBottomColor": "#BEB5B2",
                "borderBottomWidth": 0.5,
                "margin": 7,
                "width": 100,
              }
            }
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                  "marginTop": 20,
                },
              ]
            }
          >
            Or Log In with
          </Text>
          <View
            style={
              Object {
                "borderBottomColor": "#BEB5B2",
                "borderBottomWidth": 0.5,
                "margin": 7,
                "width": 100,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 40,
            "marginHorizontal": 30,
            "marginTop": "8%",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#445EA9",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 0,
              "paddingTop": 0,
              "width": "50%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
              Facebook
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "backgroundColor": "#C6563B",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "width": "50%",
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 30,
                  },
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                    "marginLeft": 16,
                    "marginRight": 0,
                    "paddingTop": 2,
                  },
                ],
                Object {
                  "fontFamily": "ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
            virtual={undefined}
          >
            
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "SinkinSans-200XLight",
                  "fontSize": 12,
                },
                Object {
                  "color": "white",
                  "fontSize": 12,
                },
              ]
            }
          >
              Google
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
